LV_analysis/1_get_eADAGE_LV_coverage.py

```diff
@@ -0,0 +1,328 @@
+# ---
+# jupyter:
+#   jupytext:
+#     formats: ipynb,py
+#     text_representation:
+#       extension: .py
+#       format_name: light
+#       format_version: '1.5'
+#       jupytext_version: 1.9.1+dev
+#   kernelspec:
+#     display_name: Python [conda env:generic_expression] *
+#     language: python
+#     name: conda-env-generic_expression-py
+# ---
+
+# # Coverage of eADAGE LV
+#
+# The goal of this notebook is to examine why genes were found to be generic. Specifically, this notebook is trying to answer the question: Are generic genes found in more multiplier latent variables compared to specific genes?
+#
+# The eADAGE model uses a DAE to extracts patterns of gene expression activity in the latent variables (referred to as nodes in the paper). Here we are examining the coverage of generic genes within these latent variables.
+#
+# **Definitions:**
+# * Generic genes: Are genes that are consistently differentially expressed across multiple simulated experiments.
+#
+# * Other genes: These are all other non-generic genes. These genes include those that are not consistently differentially expressed across simulated experiments - i.e. the genes are specifically changed in an experiment. It could also indicate genes that are consistently unchanged (i.e. housekeeping genes)
+
+# +
+# %load_ext autoreload
+# %autoreload 2
+
+import os
+import random
+import textwrap
+import scipy
+import pandas as pd
+import seaborn as sns
+import matplotlib.pyplot as plt
+from sklearn.preprocessing import MinMaxScaler
+
+import rpy2.robjects as ro
+from rpy2.robjects import pandas2ri
+from rpy2.robjects.conversion import localconverter
+
+from ponyo import utils
+from generic_expression_patterns_modules import lv
+
+# +
+# Get data directory containing gene summary data
+base_dir = os.path.abspath(os.path.join(os.getcwd(), "../"))
+data_dir = os.path.join(base_dir, "pseudomonas_analysis")
+
+# Read in config variables
+config_filename = os.path.abspath(
+    os.path.join(base_dir, "configs", "config_pseudomonas_33245.tsv")
+)
+
+params = utils.read_config(config_filename)
+
+local_dir = params["local_dir"]
+
+project_id = params["project_id"]
+# -
+
+# Output file
+nonzero_figure_filename = "nonzero_LV_coverage_eADAGE_pa.svg"
+highweight_figure_filename = "highweight_LV_coverage_eADAGE_pa.svg"
+
+# ## Load data
+
+# Get gene summary file
+summary_data_filename = os.path.join(
+    data_dir, f"generic_gene_summary_{project_id}_cbrB_v_WT.tsv"
+)
+
+# +
+# Load gene summary data
+data = pd.read_csv(summary_data_filename, sep="\t", index_col=0, header=0)
+
+# Check that genes are unique since we will be using them as dictionary keys below
+assert data.shape[0] == len(data["Gene ID"].unique())
+# -
+
+# **Manual steps to process eADAGE data**
+# 1. Data downloaded from https://zenodo.org/record/580093
```

Comment:
Thanks for the tip! I'll definitely keep this in mind for the future. This is something that just needs to be run once but in general I still like the idea of having a script to do this to try to make things more reproducible as possible. Since we're trying to wrap up the analyses for this manuscript so I don't think I'll add this for this repo but will keep it in mind for the future. Thank you!